tests/codegen/paginator-tests/build.gradle.kts

```diff
@@ -19,82 +18,56 @@ buildscript {
-            val platform = name.substring(0, name.length - 4)
-            kotlin.srcDir("$platform/$srcDir")
-            resources.srcDir("$platform/${resourcesPrefix}resources")
-            languageSettings.progressiveMode = true
-            optinAnnotations.forEach { languageSettings.optIn(it) }
-        }
-
-        val coroutinesVersion: String by project
-        val sdkVersion: String by project
-        commonMain {
-            dependencies {
-                implementation("org.jetbrains.kotlinx:kotlinx-coroutines-core:$coroutinesVersion")
-                implementation(project(":runtime:runtime-core"))
-            }
-        }
-        val jvmMain by getting {
-            kotlin.srcDir("${project.buildDir}/generated-src/src")
-        }
-    }
+kotlin.sourceSets.getByName("main") {
+    kotlin.srcDir("${project.buildDir}/generated-src/src")
+    kotlin.srcDir("${project.buildDir}/smithyprojections/paginator-tests/paginator-tests/kotlin-codegen")
 }
 
-tasks["smithyBuildJar"].enabled = false
+val coroutinesVersion: String by project
+val sdkVersion: String by project
 
-val codegen by configurations.creating
+tasks["smithyBuildJar"].enabled = false
 
+val codegen: Configuration by configurations.creating
 val generateSdk = tasks.create<SmithyBuild>("generateSdk") {
     group = "codegen"
     classpath = configurations.getByName("codegen")
     inputs.file(projectDir.resolve("smithy-build.json"))
     inputs.files(configurations.getByName("codegen"))
 }
 
-data class CodegenSourceInfo(val name: String) {
-    val projectionRootDir: File
-        get() = project.file("${project.buildDir}/smithyprojections/${project.name}/$name/kotlin-codegen/src/main/kotlin").absoluteFile
-
-    val sourceSetRootDir: File
-        get() = project.file("${project.buildDir}/generated-src/src").absoluteFile
-}
-
-val codegenSourceInfo = listOf("paginator-tests").map{ CodegenSourceInfo(it) }
-
 val stageGeneratedSources = tasks.register("stageGeneratedSources") {
     group = "codegen"
     dependsOn(generateSdk)
-    doLast {
-        codegenSourceInfo.forEach {
-            copy {
-                from("${it.projectionRootDir}")
-                into("${it.sourceSetRootDir}")
-            }
-        }
-    }
 }
 
 tasks.withType<org.jetbrains.kotlin.gradle.tasks.KotlinCompile>{
     dependsOn(stageGeneratedSources)
 }
 
-val smithyCliConfig = configurations.maybeCreate("smithyCli")
-val smithyVersion: String by project
+tasks.test {
+    useJUnitPlatform()
+    testLogging {
+        events("passed", "skipped", "failed")
+        showStandardStreams = true
+    }
+}
+
 dependencies {
+    val smithyCliConfig = configurations.maybeCreate("smithyCli")
+    val kotlinVersion: String by project
+    val smithyVersion: String by project
+
+    implementation("org.jetbrains.kotlinx:kotlinx-coroutines-core:$coroutinesVersion")
+
     smithyCliConfig("software.amazon.smithy:smithy-cli:$smithyVersion")
```

Comment:
does this work here? 

This might need to happen in a `buildscript` block to be picked up appropriately. You can tell if the output on the commandline says something like `using explicitly configured Smithy CLI` instead of `detected ...`

Reply:
Hmm, well it worked locally.  In looking at output when executing a test I see neither of those strings.  I do see the following regardless if `smithyCliConfig` is in`buildscript {}` or not:

```
> Configure project :tests:codegen:paginator-tests
(scanned and found a Smithy CLI version 1.13.1
```

In any case, I've moved this code into the `buildscript` block as suggested as it appears more correct.